src/library.js

```diff
@@ -1547,7 +1547,37 @@ LibraryManager.library = {
 #endif
     // void *dlopen(const char *file, int mode);
     // http://pubs.opengroup.org/onlinepubs/009695399/functions/dlopen.html
-    filename = filename === 0 ? '__self__' : (ENV['LD_LIBRARY_PATH'] || '/') + Pointer_stringify(filename);
+    var searchpaths = [];
+    if (filename === 0) {
+      filename = '__self__';
+    } else {
+      var strfilename = Pointer_stringify(filename);
+      if (strfilename.length > 0 && strfilename.substr(0,1) == '/') {
+        // case of aboslute path
+        filename = strfilename;
+      } else {
+        // case of relative path
+	var isValidFile = function (filename) {
+          var target = FS.findObject(filename);
```

Comment:
looks like wrong indentation here and in a few cases below. should be 2 spaces everywhere